diff --git a/src/corelib/kernel/qcore_mac.mm b/src/corelib/kernel/qcore_mac.mm
index 5100a07a..045d23ed 100644
--- a/src/corelib/kernel/qcore_mac.mm
+++ b/src/corelib/kernel/qcore_mac.mm
@@ -322,9 +322,12 @@ @implementation QT_MANGLE_NAMESPACE(QMacAutoReleasePoolTracker)
 #ifdef Q_OS_MACOS
 bool qt_mac_applicationIsInDarkMode()
 {
-    auto appearance = [NSApp.effectiveAppearance bestMatchFromAppearancesWithNames:
-            @[ NSAppearanceNameAqua, NSAppearanceNameDarkAqua ]];
-    return [appearance isEqualToString:NSAppearanceNameDarkAqua];
+    if (@available(macOS 10.14, *)) {
+        auto appearance = [NSApp.effectiveAppearance bestMatchFromAppearancesWithNames:
+                @[ NSAppearanceNameAqua, NSAppearanceNameDarkAqua ]];
+        return [appearance isEqualToString:NSAppearanceNameDarkAqua];
+    }
+    return false;
 }
 
 bool qt_mac_runningUnderRosetta()
@@ -382,6 +385,7 @@ bool qt_mac_runningUnderRosetta()
 void qt_mac_ensureResponsible()
 {
 #if !defined(QT_APPLE_NO_PRIVATE_APIS)
+    if (@available(macOS 10.14, *)) {
     if (!responsibility_get_pid_responsible_for_pid || !responsibility_spawnattrs_setdisclaim)
         return;
 
@@ -414,6 +418,7 @@ void qt_mac_ensureResponsible()
     char **argv = *_NSGetArgv();
     posix_spawnp(&pid, argv[0], nullptr, &attr, argv, environ);
     posix_spawnattr_destroy(&attr);
+    }
 #endif
 }
 #endif // QT_BUILD_INTERNAL
