diff --git a/src/gui/rhi/qrhimetal.mm b/src/gui/rhi/qrhimetal.mm
index 24674e48..a103071d 100644
--- a/src/gui/rhi/qrhimetal.mm
+++ b/src/gui/rhi/qrhimetal.mm
@@ -2777,7 +2777,14 @@ static inline MTLPixelFormat toMetalTextureFormat(QRhiTexture::Format format, QR
         }
         desc.textureType = MTLTextureType2DArray;
 #else
-        desc.textureType = samples > 1 ? MTLTextureType2DMultisampleArray : MTLTextureType2DArray;
+        if (samples > 1) {
+            if (@available(macOS 10.14, *))
+                desc.textureType = MTLTextureType2DMultisampleArray;
+            else
+                desc.textureType = MTLTextureType2DArray;
+        } else {
+            desc.textureType = MTLTextureType2DArray;
+        }
 #endif
     } else {
         desc.textureType = samples > 1 ? MTLTextureType2DMultisample : MTLTextureType2D;
