diff --git a/src/gui/platform/darwin/qapplekeymapper.mm b/src/gui/platform/darwin/qapplekeymapper.mm
index f0fd48a46e5..c8380129ccb 100644
--- a/src/gui/platform/darwin/qapplekeymapper.mm
+++ b/src/gui/platform/darwin/qapplekeymapper.mm
@@ -454,20 +454,22 @@ static int toKeyCode(const QChar &key, int virtualKey, int modifiers)
             carbonUnicodeKey = QChar(unicodeString[0]);
 
         if (canMapCocoaEvent) {
-            // Until we've verified that the Cocoa API works as expected
-            // we first run the event through the Carbon APIs and then
-            // compare the results to Cocoa.
-            auto cocoaModifiers = toCocoaModifiers(qtModifiers);
-            auto *charactersWithModifiers = [NSApp.currentEvent charactersByApplyingModifiers:cocoaModifiers];
-
-            QChar cocoaUnicodeKey;
-            if (charactersWithModifiers.length > 0)
-                cocoaUnicodeKey = QChar([charactersWithModifiers characterAtIndex:0]);
-
-            if (cocoaUnicodeKey != carbonUnicodeKey) {
-                qCWarning(lcQpaKeyMapper) << "Mismatch between Cocoa" << cocoaUnicodeKey
-                    << "and Carbon" << carbonUnicodeKey << "for virtual key" << virtualKey
-                    << "with" << qtModifiers;
+            if (@available(macOS 10.15, *)) {
+                // Until we've verified that the Cocoa API works as expected
+                // we first run the event through the Carbon APIs and then
+                // compare the results to Cocoa.
+                auto cocoaModifiers = toCocoaModifiers(qtModifiers);
+                auto *charactersWithModifiers = [NSApp.currentEvent charactersByApplyingModifiers:cocoaModifiers];
+
+                QChar cocoaUnicodeKey;
+                if (charactersWithModifiers.length > 0)
+                    cocoaUnicodeKey = QChar([charactersWithModifiers characterAtIndex:0]);
+
+                if (cocoaUnicodeKey != carbonUnicodeKey) {
+                    qCWarning(lcQpaKeyMapper) << "Mismatch between Cocoa" << cocoaUnicodeKey
+                        << "and Carbon" << carbonUnicodeKey << "for virtual key" << virtualKey
+                        << "with" << qtModifiers;
+                }
             }
         }
 
