diff --git a/src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm b/src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm
index 044a2826..4a916d19 100644
--- a/src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm
+++ b/src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm
@@ -307,7 +307,7 @@ - (BOOL)panel:(id)sender validateURL:(NSURL *)url error:(NSError * _Nullable *)o
                         fileInfo.absoluteDir().dirName().toNSString()];
 
             auto *replaceButton = [alert addButtonWithTitle:qt_mac_AppKitString(@"SavePanel", @"Replace")];
-            replaceButton.hasDestructiveAction = YES;
+            if (@available(macOS 11.0, *)) replaceButton.hasDestructiveAction = YES;
             replaceButton.tag = 1337;
             [alert addButtonWithTitle:qt_mac_AppKitString(@"Common", @"Cancel")];
 
@@ -458,11 +458,13 @@ - (void)updateProperties
     // the save panel to update and remove the extension.
     const bool nameFieldHasExtension = m_panel.nameFieldStringValue.pathExtension.length > 0;
     if (!m_panel.allowedFileTypes && !nameFieldHasExtension && !openpanel_cast(m_panel)) {
-        if (!UTTypeDirectory.preferredFilenameExtension) {
-            m_panel.allowedContentTypes = @[ UTTypeDirectory ];
-            m_panel.allowedFileTypes = nil;
-        } else {
-            qWarning() << "UTTypeDirectory unexpectedly reported an extension";
+        if (@available(macOS 11.0, *)) {
+            if (!UTTypeDirectory.preferredFilenameExtension) {
+                m_panel.allowedContentTypes = @[ UTTypeDirectory ];
+                m_panel.allowedFileTypes = nil;
+            } else {
+                qWarning() << "UTTypeDirectory unexpectedly reported an extension";
+            }
         }
     }
 
