diff --git a/src/gui/rhi/qrhimetal.mm b/src/gui/rhi/qrhimetal.mm
index 4c618d29..1fbb094e 100644
--- a/src/gui/rhi/qrhimetal.mm
+++ b/src/gui/rhi/qrhimetal.mm
@@ -536,18 +536,21 @@ bool QRhiMetalData::setupBinaryArchive(NSURL *sourceFileUrl)
     return false;
 #endif

-    [binArch release];
-    MTLBinaryArchiveDescriptor *binArchDesc = [MTLBinaryArchiveDescriptor new];
-    binArchDesc.url = sourceFileUrl;
-    NSError *err = nil;
-    binArch = [dev newBinaryArchiveWithDescriptor: binArchDesc error: &err];
-    [binArchDesc release];
-    if (!binArch) {
-        const QString msg = QString::fromNSString(err.localizedDescription);
-        qWarning("newBinaryArchiveWithDescriptor failed: %s", qPrintable(msg));
-        return false;
+    if (@available(macOS 11.0, iOS 14.0, *)) {
+        [binArch release];
+        MTLBinaryArchiveDescriptor *binArchDesc = [MTLBinaryArchiveDescriptor new];
+        binArchDesc.url = sourceFileUrl;
+        NSError *err = nil;
+        binArch = [dev newBinaryArchiveWithDescriptor: binArchDesc error: &err];
+        [binArchDesc release];
+        if (!binArch) {
+            const QString msg = QString::fromNSString(err.localizedDescription);
+            qWarning("newBinaryArchiveWithDescriptor failed: %s", qPrintable(msg));
+            return false;
+        }
+        return true;
     }
-    return true;
+    return false;
 }

 bool QRhiMetal::create(QRhi::Flags flags)
@@ -579,19 +582,21 @@ bool QRhiMetal::create(QRhi::Flags flags)
     // apparently change when the system is rebooted.

 #ifdef Q_OS_MACOS
-    const MTLDeviceLocation deviceLocation = [d->dev location];
-    switch (deviceLocation) {
-    case MTLDeviceLocationBuiltIn:
-        driverInfoStruct.deviceType = QRhiDriverInfo::IntegratedDevice;
-        break;
-    case MTLDeviceLocationSlot:
-        driverInfoStruct.deviceType = QRhiDriverInfo::DiscreteDevice;
-        break;
-    case MTLDeviceLocationExternal:
-        driverInfoStruct.deviceType = QRhiDriverInfo::ExternalDevice;
-        break;
-    default:
-        break;
+    if (@available(macOS 10.15, *)) {
+        const MTLDeviceLocation deviceLocation = [d->dev location];
+        switch (deviceLocation) {
+        case MTLDeviceLocationBuiltIn:
+            driverInfoStruct.deviceType = QRhiDriverInfo::IntegratedDevice;
+            break;
+        case MTLDeviceLocationSlot:
+            driverInfoStruct.deviceType = QRhiDriverInfo::DiscreteDevice;
+            break;
+        case MTLDeviceLocationExternal:
+            driverInfoStruct.deviceType = QRhiDriverInfo::ExternalDevice;
+            break;
+        default:
+            break;
+        }
     }
 #else
     driverInfoStruct.deviceType = QRhiDriverInfo::IntegratedDevice;
@@ -606,18 +611,21 @@ bool QRhiMetal::create(QRhi::Flags flags)
     else
         d->cmdQueue = [d->dev newCommandQueue];

-    d->captureMgr = [MTLCaptureManager sharedCaptureManager];
-    // Have a custom capture scope as well which then shows up in XCode as
-    // an option when capturing, and becomes especially useful when having
-    // multiple windows with multiple QRhis.
-    d->captureScope = [d->captureMgr newCaptureScopeWithCommandQueue: d->cmdQueue];
-    const QString label = QString::asprintf("Qt capture scope for QRhi %p", this);
-    d->captureScope.label = label.toNSString();
+    if (@available(macOS 10.15, iOS 13.0, *)) {
+        d->captureMgr = [MTLCaptureManager sharedCaptureManager];
+        d->captureScope = [d->captureMgr newCaptureScopeWithCommandQueue: d->cmdQueue];
+        const QString label = QString::asprintf("Qt capture scope for QRhi %p", this);
+        d->captureScope.label = label.toNSString();
+    } else {
+        d->captureMgr = nil;
+        d->captureScope = nil;
+    }

 #if defined(Q_OS_MACOS) || defined(Q_OS_VISIONOS)
     caps.maxTextureSize = 16384;
     caps.baseVertexAndInstance = true;
-    caps.isAppleGPU = [d->dev supportsFamily:MTLGPUFamilyApple7];
+    if (@available(macOS 10.15, *))
+        caps.isAppleGPU = [d->dev supportsFamily:MTLGPUFamilyApple7];
     caps.maxThreadGroupSize = 1024;
     caps.multiView = true;
 #elif defined(Q_OS_TVOS)
@@ -651,8 +659,10 @@ bool QRhiMetal::create(QRhi::Flags flags)
             caps.supportedSampleCounts.append(sampleCount);
     }

-    caps.shadingRateMap = [d->dev supportsRasterizationRateMapWithLayerCount: 1];
-    if (caps.shadingRateMap && caps.multiView)
-        caps.shadingRateMap = [d->dev supportsRasterizationRateMapWithLayerCount: 2];
+    if (@available(macOS 12.0, iOS 15.0, *)) {
+        caps.shadingRateMap = [d->dev supportsRasterizationRateMapWithLayerCount: 1];
+        if (caps.shadingRateMap && caps.multiView)
+            caps.shadingRateMap = [d->dev supportsRasterizationRateMapWithLayerCount: 2];
+    }

     if (rhiFlags.testFlag(QRhi::EnablePipelineCacheDataSave))
         d->setupBinaryArchive();
@@ -970,6 +980,10 @@ QByteArray QRhiMetal::pipelineCacheData()
     if (!d->binArch || !rhiFlags.testFlag(QRhi::EnablePipelineCacheDataSave))
         return data;

+    if (@available(macOS 11.0, iOS 14.0, *)) { } else {
+        return data;
+    }
+
     QTemporaryFile tmp;
     if (!tmp.open()) {
         qCDebug(QRHI_LOG_INFO, "pipelineCacheData: Failed to create temporary file for Metal");
@@ -1875,10 +1889,12 @@ static inline void rebindShaderResources(QMetalCommandBuffer *cbD, int resourceS
     // This is specifically needed for visionOS.
     if (cbD->currentTarget->resourceType() == QRhiResource::TextureRenderTarget) {
         QRhiTextureRenderTarget *rt = static_cast<QRhiTextureRenderTarget *>(cbD->currentTarget);
-        if (QRhiShadingRateMap *srm = rt->description().shadingRateMap()) {
-            if (id<MTLRasterizationRateMap> rateMap = QRHI_RES(QMetalShadingRateMap, srm)->d->rateMap) {
-                auto screenSize = [rateMap screenSize];
-                outputSize = QSize(screenSize.width, screenSize.height);
+        if (@available(macOS 12.0, iOS 15.0, *)) {
+            if (QRhiShadingRateMap *srm = rt->description().shadingRateMap()) {
+                if (id<MTLRasterizationRateMap> rateMap = QRHI_RES(QMetalShadingRateMap, srm)->d->rateMap) {
+                    auto screenSize = [rateMap screenSize];
+                    outputSize = QSize(screenSize.width, screenSize.height);
+                }
             }
         }
     }
@@ -2442,7 +2458,7 @@ static void endTessellationComputeEncoding(QMetalCommandBuffer *cbD)

     __block int thisFrameSlot = currentFrameSlot;
     [commandBuffer addCompletedHandler: ^(id<MTLCommandBuffer> cb) {
-        swapChainD->d->lastGpuTime[thisFrameSlot] += cb.GPUEndTime - cb.GPUStartTime;
+        swapChainD->d->lastGpuTime[thisFrameSlot] += ([&]() -> double { if (@available(macOS 10.15, *)) return cb.GPUEndTime - cb.GPUStartTime; return 0; }());
         dispatch_semaphore_signal(swapChainD->d->sem[thisFrameSlot]);
     }];

@@ -2566,7 +2582,7 @@ static void endTessellationComputeEncoding(QMetalCommandBuffer *cbD)
     // offscreen frames wait for completion, unlike swapchain ones
     [cb waitUntilCompleted];

-    d->ofr.lastGpuTime += cb.GPUEndTime - cb.GPUStartTime;
+    d->ofr.lastGpuTime += ([&]() -> double { if (@available(macOS 10.15, *)) return cb.GPUEndTime - cb.GPUStartTime; return 0; }());

     finishActiveReadbacks(true);

@@ -2608,10 +2624,10 @@ static void endTessellationComputeEncoding(QMetalCommandBuffer *cbD)

     if (inFrame) {
         if (d->ofr.active) {
-            d->ofr.lastGpuTime += cb.GPUEndTime - cb.GPUStartTime;
+            d->ofr.lastGpuTime += ([&]() -> double { if (@available(macOS 10.15, *)) return cb.GPUEndTime - cb.GPUStartTime; return 0; }());
             d->ofr.cbWrapper.d->cb = d->newCommandBuffer();
         } else {
-            swapChainD->d->lastGpuTime[currentFrameSlot] += cb.GPUEndTime - cb.GPUStartTime;
+            swapChainD->d->lastGpuTime[currentFrameSlot] += ([&]() -> double { if (@available(macOS 10.15, *)) return cb.GPUEndTime - cb.GPUStartTime; return 0; }());
             swapChainD->cbWrapper.d->cb = d->newCommandBuffer();
         }
     }
@@ -2649,8 +2665,10 @@ static void endTessellationComputeEncoding(QMetalCommandBuffer *cbD)
         rp.stencilAttachment.clearStencil = depthStencilClearValue.stencilClearValue();
     }

-    if (shadingRateMap)
-        rp.rasterizationRateMap = QRHI_RES(QMetalShadingRateMap, shadingRateMap)->d->rateMap;
+    if (@available(macOS 12.0, iOS 15.0, *)) {
+        if (shadingRateMap)
+            rp.rasterizationRateMap = QRHI_RES(QMetalShadingRateMap, shadingRateMap)->d->rateMap;
+    }

     return rp;
 }
@@ -3766,9 +3784,15 @@ static inline MTLPixelFormat toMetalTextureFormat(QRhiTexture::Format format, QR
     switch (m_type) {
     case DepthStencil:
 #ifdef Q_OS_MACOS
-        if (rhiD->caps.isAppleGPU) {
-            desc.storageMode = MTLStorageModeMemoryless;
-            d->format = MTLPixelFormatDepth32Float_Stencil8;
+        if (@available(macOS 11.0, *)) {
+            if (rhiD->caps.isAppleGPU) {
+                desc.storageMode = MTLStorageModeMemoryless;
+                d->format = MTLPixelFormatDepth32Float_Stencil8;
+            } else {
+                desc.storageMode = MTLStorageModePrivate;
+                d->format = rhiD->d->dev.depth24Stencil8PixelFormatSupported
+                        ? MTLPixelFormatDepth24Unorm_Stencil8 : MTLPixelFormatDepth32Float_Stencil8;
+            }
         } else {
             desc.storageMode = MTLStorageModePrivate;
             d->format = rhiD->d->dev.depth24Stencil8PixelFormatSupported
@@ -3946,7 +3970,10 @@ static inline MTLPixelFormat toMetalTextureFormat(QRhiTexture::Format format, QR
     } else if (is1D) {
         desc.textureType = isArray ? MTLTextureType1DArray : MTLTextureType1D;
     } else if (isArray) {
-        desc.textureType = samples > 1 ? MTLTextureType2DMultisampleArray : MTLTextureType2DArray;
+        if (@available(macOS 10.14, iOS 14.0, *))
+            desc.textureType = samples > 1 ? MTLTextureType2DMultisampleArray : MTLTextureType2DArray;
+        else
+            desc.textureType = MTLTextureType2DArray;
     } else {
         desc.textureType = samples > 1 ? MTLTextureType2DMultisample : MTLTextureType2D;
     }
@@ -5096,19 +5123,23 @@ static inline MTLLanguageVersion toMetalLanguageVersion(const QShaderVersion &ve

 void QRhiMetalData::trySeedingRenderPipelineFromBinaryArchive(MTLRenderPipelineDescriptor *rpDesc)
 {
-    if (binArch)  {
-        NSArray *binArchArray = [NSArray arrayWithObjects: binArch, nil];
-        rpDesc.binaryArchives = binArchArray;
+    if (@available(macOS 11.0, iOS 14.0, *)) {
+        if (binArch)  {
+            NSArray *binArchArray = [NSArray arrayWithObjects: binArch, nil];
+            rpDesc.binaryArchives = binArchArray;
+        }
     }
 }

 void QRhiMetalData::addRenderPipelineToBinaryArchive(MTLRenderPipelineDescriptor *rpDesc)
 {
-    if (binArch) {
-        NSError *err = nil;
-        if (![binArch addRenderPipelineFunctionsWithDescriptor: rpDesc error: &err]) {
-            const QString msg = QString::fromNSString(err.localizedDescription);
-            qWarning("Failed to collect render pipeline functions to binary archive: %s", qPrintable(msg));
+    if (@available(macOS 11.0, iOS 14.0, *)) {
+        if (binArch) {
+            NSError *err = nil;
+            if (![binArch addRenderPipelineFunctionsWithDescriptor: rpDesc error: &err]) {
+                const QString msg = QString::fromNSString(err.localizedDescription);
+                qWarning("Failed to collect render pipeline functions to binary archive: %s", qPrintable(msg));
+            }
         }
     }
 }
@@ -6078,19 +6109,23 @@ static inline bool matches(const QShaderDescription::InOutVariable &a, const QSh

 void QRhiMetalData::trySeedingComputePipelineFromBinaryArchive(MTLComputePipelineDescriptor *cpDesc)
 {
-    if (binArch) {
-        NSArray *binArchArray = [NSArray arrayWithObjects: binArch, nil];
-        cpDesc.binaryArchives = binArchArray;
+    if (@available(macOS 11.0, iOS 14.0, *)) {
+        if (binArch) {
+            NSArray *binArchArray = [NSArray arrayWithObjects: binArch, nil];
+            cpDesc.binaryArchives = binArchArray;
+        }
     }
 }

 void QRhiMetalData::addComputePipelineToBinaryArchive(MTLComputePipelineDescriptor *cpDesc)
 {
-    if (binArch) {
-        NSError *err = nil;
-        if (![binArch addComputePipelineFunctionsWithDescriptor: cpDesc error: &err]) {
-            const QString msg = QString::fromNSString(err.localizedDescription);
-            qWarning("Failed to collect compute pipeline functions to binary archive: %s", qPrintable(msg));
+    if (@available(macOS 11.0, iOS 14.0, *)) {
+        if (binArch) {
+            NSError *err = nil;
+            if (![binArch addComputePipelineFunctionsWithDescriptor: cpDesc error: &err]) {
+                const QString msg = QString::fromNSString(err.localizedDescription);
+                qWarning("Failed to collect compute pipeline functions to binary archive: %s", qPrintable(msg));
+            }
         }
     }
 }
@@ -6602,7 +6637,7 @@ static inline bool matches(const QShaderDescription::InOutVariable &a, const QSh
         info.limits.colorComponentValue.maxColorComponentValue = screen.maximumExtendedDynamicRangeColorComponentValue;
-        info.limits.colorComponentValue.maxPotentialColorComponentValue = screen.maximumPotentialExtendedDynamicRangeColorComponentValue;
+        info.limits.colorComponentValue.maxPotentialColorComponentValue = ([&]() -> CGFloat { if (@available(macOS 10.15, *)) return screen.maximumPotentialExtendedDynamicRangeColorComponentValue; return 1.0; }());
 #elif defined(Q_OS_IOS)
         if (@available(iOS 16.0, *)) {
             UIView *view = reinterpret_cast<UIView *>(m_window->winId());
