diff --git a/src/gui/util/qgraphicsframecapturemetal.mm b/src/gui/util/qgraphicsframecapturemetal.mm
index 259872e4..7b963649 100644
--- a/src/gui/util/qgraphicsframecapturemetal.mm
+++ b/src/gui/util/qgraphicsframecapturemetal.mm
@@ -21,9 +21,10 @@

 QGraphicsFrameCaptureMetal::QGraphicsFrameCaptureMetal()
 {
-    qputenv("METAL_CAPTURE_ENABLED", QByteArrayLiteral("1"));
-
-    m_captureDescriptor = [MTLCaptureDescriptor new];
+    if (@available(macOS 10.15, iOS 13.0, *)) {
+        qputenv("METAL_CAPTURE_ENABLED", QByteArrayLiteral("1"));
+        m_captureDescriptor = [MTLCaptureDescriptor new];
+    }
 }

 QGraphicsFrameCaptureMetal::~QGraphicsFrameCaptureMetal()
@@ -42,6 +43,11 @@
     if (!rhi)
         return;

+    if (@available(macOS 10.15, iOS 13.0, *)) { } else {
+        m_initialized = false;
+        return;
+    }
+
     QRhi::Implementation backend = rhi->backend();
     const QRhiNativeHandles *nh = rhi->nativeHandles();

@@ -90,11 +96,13 @@
         return;
     }

-    updateCaptureFileName();
-    NSError *error;
-    if (![m_captureManager startCaptureWithDescriptor:m_captureDescriptor error:&error]) {
-        QString errorMsg = QString::fromNSString(error.localizedDescription);
-        qCWarning(lcGraphicsFrameCapture, "Failed to start capture : %s", qPrintable(errorMsg));
+    if (@available(macOS 10.15, iOS 13.0, *)) {
+        updateCaptureFileName();
+        NSError *error;
+        if (![m_captureManager startCaptureWithDescriptor:m_captureDescriptor error:&error]) {
+            QString errorMsg = QString::fromNSString(error.localizedDescription);
+            qCWarning(lcGraphicsFrameCapture, "Failed to start capture : %s", qPrintable(errorMsg));
+        }
     }
 }

@@ -128,7 +136,9 @@
         return false;
     }

-    return [m_captureManager isCapturing];
+    if (@available(macOS 10.15, iOS 13.0, *))
+        return [m_captureManager isCapturing];
+    return false;
 }

 void QGraphicsFrameCaptureMetal::openCapture()
@@ -158,15 +168,15 @@

 void QGraphicsFrameCaptureMetal::updateCaptureFileName()
 {
-    m_traceURL = QUrl::fromLocalFile(m_capturePath + u"/" + m_capturePrefix + u"_"
-                                     + QString::number(frameNumber) + u".gputrace")
-                         .toNSURL();
-    // We need to remove the trace file if it already existed else MTLCaptureManager
-    // will fail to.
-    if ([NSFileManager.defaultManager fileExistsAtPath:m_traceURL.path])
-        [NSFileManager.defaultManager removeItemAtURL:m_traceURL error:nil];
-
-    m_captureDescriptor.outputURL = m_traceURL;
+    if (@available(macOS 10.15, iOS 13.0, *)) {
+        m_traceURL = QUrl::fromLocalFile(m_capturePath + u"/" + m_capturePrefix + u"_"
+                                         + QString::number(frameNumber) + u".gputrace")
+                             .toNSURL();
+        if ([NSFileManager.defaultManager fileExistsAtPath:m_traceURL.path])
+            [NSFileManager.defaultManager removeItemAtURL:m_traceURL error:nil];
+
+        m_captureDescriptor.outputURL = m_traceURL;
+    }
 }

 QT_END_NAMESPACE
