diff --git a/src/plugins/platforms/cocoa/qcocoadrag.mm b/src/plugins/platforms/cocoa/qcocoadrag.mm
index 80c1559f422..4aa9bc884c1 100644
--- a/src/plugins/platforms/cocoa/qcocoadrag.mm
+++ b/src/plugins/platforms/cocoa/qcocoadrag.mm
@@ -170,7 +170,7 @@
     for (NSPasteboardItem *item in dragBoard.pasteboardItems) {
         bool isUrl = false;
         for (NSPasteboardType type in item.types) {
-            if ([type isEqualToString:UTTypeFileURL.identifier]) {
+            if ([type isEqualToString:@"public.file-url"]) {
                 isUrl = true;
                 break;
             }
