diff --git a/src/plugins/platforms/cocoa/qcocoatheme.mm b/src/plugins/platforms/cocoa/qcocoatheme.mm
index a5d45ef1623..0f076e0f5fe 100644
--- a/src/plugins/platforms/cocoa/qcocoatheme.mm
+++ b/src/plugins/platforms/cocoa/qcocoatheme.mm
@@ -68,7 +68,11 @@
     // System palette initialization:
     QBrush br = qt_mac_toQBrush([NSColor selectedControlColor]);
     palette->setBrush(QPalette::Active, QPalette::Highlight, br);
-    const auto inactiveHighlight = qt_mac_toQBrush([NSColor unemphasizedSelectedContentBackgroundColor]);
+    QBrush inactiveHighlight;
+    if (@available(macOS 10.14, *))
+        inactiveHighlight = qt_mac_toQBrush([NSColor unemphasizedSelectedContentBackgroundColor]);
+    else
+        inactiveHighlight = qt_mac_toQBrush([NSColor secondarySelectedControlColor]);
     palette->setBrush(QPalette::Inactive, QPalette::Highlight, inactiveHighlight);
     palette->setBrush(QPalette::Disabled, QPalette::Highlight, inactiveHighlight);
 
@@ -98,7 +102,10 @@
     palette->setColor(QPalette::Inactive, QPalette::PlaceholderText, qc);
     palette->setColor(QPalette::Disabled, QPalette::PlaceholderText, qc);
 
-    qc = qt_mac_toQColor([NSColor controlAccentColor]);
+    if (@available(macOS 10.14, *))
+        qc = qt_mac_toQColor([NSColor controlAccentColor]);
+    else
+        qc = qt_mac_toQColor([NSColor selectedControlColor]);
     palette->setColor(QPalette::Accent, qc);
 
     return palette;
@@ -159,7 +166,12 @@ inline QMacPaletteMap(QPlatformTheme::Palette p, NSColor *a, NSColor *i) :
         if (mac_widget_colors[i].paletteRole == QPlatformTheme::MenuPalette
                 || mac_widget_colors[i].paletteRole == QPlatformTheme::MenuBarPalette) {
             // Cheap approximation for NSVisualEffectView (see deprecation note for selectedMenuItemTextColor)
-            auto selectedMenuItemColor = [[NSColor controlAccentColor] highlightWithLevel:0.3];
+            NSColor *accentBase = nil;
+            if (@available(macOS 10.14, *))
+                accentBase = [NSColor controlAccentColor];
+            else
+                accentBase = [NSColor selectedControlColor];
+            auto selectedMenuItemColor = [accentBase highlightWithLevel:0.3];
             pal.setBrush(QPalette::Highlight, qt_mac_toQColor(selectedMenuItemColor));
             qc = qt_mac_toQColor([NSColor labelColor]);
             pal.setBrush(QPalette::ButtonText, qc);
@@ -178,16 +190,19 @@ inline QMacPaletteMap(QPlatformTheme::Palette p, NSColor *a, NSColor *i) :
             pal.setColor(QPalette::Active, QPalette::ButtonText,
                          pal.color(QPalette::Active, QPalette::Text));
         } else if (mac_widget_colors[i].paletteRole == QPlatformTheme::ItemViewPalette) {
-            NSArray<NSColor *> *baseColors = nil;
-            NSColor *activeHighlightColor = nil;
-            baseColors = [NSColor alternatingContentBackgroundColors];
-            activeHighlightColor = [NSColor selectedContentBackgroundColor];
-            pal.setBrush(QPalette::Inactive, QPalette::HighlightedText,
-                         qt_mac_toQBrush([NSColor unemphasizedSelectedTextColor]));
-            pal.setBrush(QPalette::Base, qt_mac_toQBrush(baseColors[0]));
-            pal.setBrush(QPalette::AlternateBase, qt_mac_toQBrush(baseColors[1]));
-            pal.setBrush(QPalette::Active, QPalette::Highlight,
-                         qt_mac_toQBrush(activeHighlightColor));
+            if (@available(macOS 10.14, *)) {
+                NSArray<NSColor *> *baseColors = [NSColor alternatingContentBackgroundColors];
+                NSColor *activeHighlightColor = [NSColor selectedContentBackgroundColor];
+                pal.setBrush(QPalette::Inactive, QPalette::HighlightedText,
+                             qt_mac_toQBrush([NSColor unemphasizedSelectedTextColor]));
+                pal.setBrush(QPalette::Base, qt_mac_toQBrush(baseColors[0]));
+                pal.setBrush(QPalette::AlternateBase, qt_mac_toQBrush(baseColors[1]));
+                pal.setBrush(QPalette::Active, QPalette::Highlight,
+                             qt_mac_toQBrush(activeHighlightColor));
+            } else {
+                pal.setBrush(QPalette::Active, QPalette::Highlight,
+                             qt_mac_toQBrush([NSColor alternateSelectedControlColor]));
+            }
             pal.setBrush(QPalette::Active, QPalette::HighlightedText,
                          qt_mac_toQBrush([NSColor alternateSelectedControlTextColor]));
             pal.setBrush(QPalette::Inactive, QPalette::Text,
@@ -300,16 +315,24 @@ inline QMacPaletteMap(QPlatformTheme::Palette p, NSColor *a, NSColor *i) :
 
     if (type == SystemPalette) {
         if (!m_systemPalette) {
-            [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:^{
+            if (@available(macOS 11.0, *)) {
+                [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:^{
+                    m_systemPalette = qt_mac_createSystemPalette();
+                }];
+            } else {
                 m_systemPalette = qt_mac_createSystemPalette();
-            }];
+            }
         }
         return m_systemPalette;
     } else {
         if (m_palettes.isEmpty()) {
-            [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:^{
+            if (@available(macOS 11.0, *)) {
+                [NSApp.effectiveAppearance performAsCurrentDrawingAppearance:^{
+                    m_palettes = qt_mac_createRolePalettes();
+                }];
+            } else {
                 m_palettes = qt_mac_createRolePalettes();
-            }];
+            }
         }
         return m_palettes.value(type, nullptr);
     }
@@ -392,8 +415,19 @@ QPixmap qt_mac_convert_iconref(const IconRef icon, int width, int height)
     case DirIcon:
     case DirClosedIcon:
     case DirLinkIcon: {
-        NSImage *icon = [NSWorkspace.sharedWorkspace iconForContentType:UTTypeFolder];
-        return qt_mac_toQPixmap(icon, size);
+        if (@available(macOS 11.0, *)) {
+            NSImage *icon = [NSWorkspace.sharedWorkspace iconForContentType:UTTypeFolder];
+            return qt_mac_toQPixmap(icon, size);
+        } else {
+            IconRef icon = nullptr;
+            QT_IGNORE_DEPRECATIONS(GetIconRef(kOnSystemDisk, kSystemIconsCreator, kGenericFolderIcon, &icon));
+            if (icon) {
+                QPixmap pixmap = qt_mac_convert_iconref(icon, size.width(), size.height());
+                QT_IGNORE_DEPRECATIONS(ReleaseIconRef(icon));
+                return pixmap;
+            }
+            return QPixmap();
+        }
     }
     case FileLinkIcon:
     case FileIcon:
@@ -480,7 +514,8 @@ QPixmap qt_mac_convert_iconref(const IconRef icon, int width, int height)
     NSAppearance *appearance = nil;
     switch (scheme) {
     case Qt::ColorScheme::Dark:
-        appearance = [NSAppearance appearanceNamed:NSAppearanceNameDarkAqua];
+        if (@available(macOS 10.14, *))
+            appearance = [NSAppearance appearanceNamed:NSAppearanceNameDarkAqua];
         break;
     case Qt::ColorScheme::Light:
         appearance = [NSAppearance appearanceNamed:NSAppearanceNameAqua];
@@ -504,10 +539,14 @@ QPixmap qt_mac_convert_iconref(const IconRef icon, int width, int height)
 */
 void QCocoaTheme::updateColorScheme()
 {
-    auto appearance = [NSApp.effectiveAppearance bestMatchFromAppearancesWithNames:
-            @[ NSAppearanceNameAqua, NSAppearanceNameDarkAqua ]];
-    m_colorScheme = [appearance isEqualToString:NSAppearanceNameDarkAqua] ?
-        Qt::ColorScheme::Dark : Qt::ColorScheme::Light;
+    if (@available(macOS 10.14, *)) {
+        auto appearance = [NSApp.effectiveAppearance bestMatchFromAppearancesWithNames:
+                @[ NSAppearanceNameAqua, NSAppearanceNameDarkAqua ]];
+        m_colorScheme = [appearance isEqualToString:NSAppearanceNameDarkAqua] ?
+            Qt::ColorScheme::Dark : Qt::ColorScheme::Light;
+    } else {
+        m_colorScheme = Qt::ColorScheme::Light;
+    }
 }
 
 Qt::ContrastPreference QCocoaTheme::contrastPreference() const
