diff --git a/src/gui/platform/darwin/qapplekeymapper.mm b/src/gui/platform/darwin/qapplekeymapper.mm
index 9166d8ba..fd890be0 100644
--- a/src/gui/platform/darwin/qapplekeymapper.mm
+++ b/src/gui/platform/darwin/qapplekeymapper.mm
@@ -455,7 +455,7 @@ static int toKeyCode(const QChar &key, int virtualKey, int modifiers)
         if (err == noErr && actualStringLength)
             carbonUnicodeKey = QChar(unicodeString[0]);

-        if (canMapCocoaEvent) {
+        if (canMapCocoaEvent) { if (@available(macOS 10.15, *)) {
             // Until we've verified that the Cocoa API works as expected
             // we first run the event through the Carbon APIs and then
             // compare the results to Cocoa.
@@ -468,7 +468,7 @@ static int toKeyCode(const QChar &key, int virtualKey, int modifiers)
                     << "and Carbon" << carbonUnicodeKey << "for virtual key" << virtualKey
                     << "with" << qtModifiers;
             }
-        }
+        } }

         int qtKey = toKeyCode(carbonUnicodeKey, virtualKey, qtModifiers);
         if (qtKey == Qt::Key_unknown)
