diff --git a/src/plugins/platforms/cocoa/qcocoaservices.mm b/src/plugins/platforms/cocoa/qcocoaservices.mm
index 87212c26..7008abe3 100644
--- a/src/plugins/platforms/cocoa/qcocoaservices.mm
+++ b/src/plugins/platforms/cocoa/qcocoaservices.mm
@@ -43,30 +43,43 @@
 class QCocoaColorPicker : public QPlatformServiceColorPicker
 {
 public:
-    QCocoaColorPicker() : m_colorSampler([NSColorSampler new]) {}
-    ~QCocoaColorPicker() { [m_colorSampler release]; }
+    QCocoaColorPicker() {
+        if (@available(macOS 10.15, *))
+            m_colorSampler = [NSColorSampler new];
+    }
+    ~QCocoaColorPicker() {
+        if (@available(macOS 10.15, *))
+            [m_colorSampler release];
+    }

     void pickColor() override
     {
-        [m_colorSampler showSamplerWithSelectionHandler:^(NSColor *selectedColor) {
-            emit colorPicked(qt_mac_toQColor(selectedColor));
-        }];
+        if (@available(macOS 10.15, *)) {
+            [m_colorSampler showSamplerWithSelectionHandler:^(NSColor *selectedColor) {
+                emit colorPicked(qt_mac_toQColor(selectedColor));
+            }];
+        }
     }
 private:
-    NSColorSampler *m_colorSampler = nullptr;
+    NSColorSampler *m_colorSampler API_AVAILABLE(macos(10.15)) = nullptr;
 };


 QPlatformServiceColorPicker *QCocoaServices::colorPicker(QWindow *parent)
 {
     Q_UNUSED(parent);
-    return new QCocoaColorPicker;
+    if (@available(macOS 10.15, *))
+        return new QCocoaColorPicker;
+    return nullptr;
 }

 bool QCocoaServices::hasCapability(Capability capability) const
 {
     switch (capability) {
-    case ColorPicking: return true;
+    case ColorPicking:
+        if (@available(macOS 10.15, *))
+            return true;
+        return false;
     default: return false;
     }
 }
