diff --git a/src/gui/platform/darwin/qappleiconengine.mm b/src/gui/platform/darwin/qappleiconengine.mm
index 84e9ff4e..9c1afa9e 100644
--- a/src/gui/platform/darwin/qappleiconengine.mm
+++ b/src/gui/platform/darwin/qappleiconengine.mm
@@ -279,7 +279,7 @@
     });
     NSString *systemIconName = it != std::end(iconMap) ? it->second : iconName.toNSString();
 #if defined(Q_OS_MACOS)
-    return [NSImage imageWithSystemSymbolName:systemIconName accessibilityDescription:nil];
+    if (@available(macOS 11.0, *)) return [NSImage imageWithSystemSymbolName:systemIconName accessibilityDescription:nil]; return nil;
 #elif defined(QT_PLATFORM_UIKIT)
     return [UIImage systemImageNamed:systemIconName];
 #endif
@@ -359,6 +359,7 @@
 #if defined(Q_OS_MACOS)
 auto *configuredImage(const NSImage *image, const QColor &color)
 {
+    if (@available(macOS 11.0, *)) {
     auto *config = [NSImageSymbolConfiguration configurationWithPointSize:48
                                                weight:NSFontWeightRegular
                                                scale:NSImageSymbolScaleLarge];
@@ -379,6 +380,8 @@
             return YES;
         }];
     return tintedImage;
+    }
+    return (NSImage *)image;
 }
 #elif defined(QT_PLATFORM_UIKIT)
 auto *configuredImage(const UIImage *image, const QColor &color)
